### PR TITLE
NEW: Travis CI build of the Documentation, and deploy to github-pages

### PR DESCRIPTION
**Pull request by Ne02ptzero**
- New `.travis.yml` file, that describes the build processus, and the deploy one
- Build will trigger on any branches, PR, commit.
- Deploy to github-pages will only happen when a commit is pushed to `master`.

Firefox preview:
2018-01-09_1919x965_scrot
Lynx preview:
2018-01-09_958x278_scrot

> Do not judge me on my HTML skill, for I am a System developer.
> - Me, right now

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+before_install:
+    - sudo apt-get -qq update
+    - sudo apt-get install -y groff
+
+script:
+    - make -C protocols/ all html
```
Comment (JetseVerschuren) [inline .travis.yml:6]: Isn't it easier to do
`mkdir build && cd build`
`make -C ../protocols/ all html`

File: theme/index.html [hunk 2]
```diff
@@ -0,0 +1,120 @@
+<html>
+    <head>
+        <title>Meshnet/Documentation</title>
+        <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-wvfXpqpZZVQGK6TAh5PVlGOfQNHSoD2xbE+QkPxCAFlNEevoEH3Sl0sibVcOQVnN" crossorigin="anonymous">
+        <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">
+        <style>
```
Comment (JetseVerschuren) [inline theme/index.html:6]: Is there a reason why the style isn't in a separate file?

**Review by JetseVerschuren — APPROVED** [review]
Overall it looks good to me.

File: theme/index.html (hunk 2 above)
Comment (0x5c) [reply theme/index.html:6]: For simplicity; if the style is needed for one page only, why would we waste requests to get resources that can easily be inside the html?

**Review by 0x5c — APPROVED** [review]
LGTM

File: .travis.yml (hunk 1 above)
Comment (Ne02ptzero) [reply .travis.yml:6]: `-C` will change directory, you still need to copy the output files after build.

**Pull request merged**